frame/composable-support/src/validation.rs

```diff
@@ -53,16 +56,50 @@ impl<T: Validate<U> + Validate<V>, U, V> Validate<(U, V)> for T {
 	}
 }
 
-impl<T: codec::Decode + Validate<(U, V)>, U, V> codec::Decode for Validated<T, (U, V)> {
+// as per substrate pattern and existing macroses for similar purposes, they tend to make things
+// flat like `#[impl_trait_for_tuples::impl_for_tuples(30)]`
+// so if we will need more than 3, can consider it
+impl<T: Validate<U> + Validate<V> + Validate<W>, U, V, W> Validate<(U, V, W)> for T {
+	#[inline(always)]
+	fn validate(self) -> Result<Self, &'static str> {
+		let value = Validate::<U>::validate(self)?;
+		let value = Validate::<V>::validate(value)?;
+		let value = Validate::<W>::validate(value)?;
+		Ok(value)
+	}
+}
```

Comment:
This is already derived if you use `(U, (V, (W, Valid))`

Reply:
I see. But with this user has 2 options for convenience? :)